### ISSUE TITLE
Views and attributes: id, type

### ISSUE THREAD

**Issue by mdonges** [structr/structr#527]
Derat Structr Team,

i just started to play with the 3.1.2 release while we are still running on an old 2.1.4...
I like that you fixed, that all references have consistently the id, type and even name attribute.

You introduced the "http://structr.server:8082/structr/rest/SomeType/custom" view over the ".../ui" view which we used previously...

In the custom view the attributes
id, and type are not included, which makes some interactions problematic: e.g.

http://structr.server:8082/structr/rest/File/custom?path=/some/cool/path/file.txt

In such cases i cannot find out which id the object has and I am being forced to do at least another call.
Or did I get something wrong here?

Is there a view where i guaranteed get all available arrtibutes and all first level relationship ids guaranteed?

In my oppinion
id
type
attributes should be on any (at least all the default) view always.

**Comment by amorgner** [comment]
Glad you like Structr 3.1!

The `ui` was always meant as an internal view and subject to changes. With the constantly growing functionality the `ui` view became bigger and bigger, and custom attributes were also automatically added to this view.

We decided to separate all built-in attributes to restore the original scope and the initial idea (namely that these attributes were used in the backend UI) and put all custom attributes into a new system view called `custom`.

To answer your questions:

> Is there a view where i guaranteed get all available arrtibutes and all first level relationship ids guaranteed?

Yes, it's the `all` view. It was there from the beginning but somewhat dangerous to use as it could potentially lead to contain the entire database (that's the risk of offering a flexible and unlimited tree-based interface to a graph database ...). But we think we tamed the beast (the `all` view) so that it can be used instead of the old `ui` view (in fact it is used in the Data section).

However, best practice is to create your own views and not rely on built-in functionality too much.
